### PR TITLE
[fix](array/map) fix resize impl in array/map

### PR DESCRIPTION
**Pull request by amorynan**
## Proposed changes
this resize function called shrink function which defined in IColumn, and function set_num_rows in  Block.cpp call shrink to make column rows changed, maybe less or more.
In this situation if we would make less rows which like cut the rows, so we should also make nested data in array/map resize to the less number of rows , otherwise we will meet exception :
```
mysql> /*set ShuffleSendBytes=0|ShuffleSendRows=0|SqlHash=c359573c8c7c33238ec34129e8ef0131|peakMemoryBytes=203840|SqlDigest=|cloudClusterName=UNKNOWN|TraceId=|WorkloadGroup=normal|FuzzyVariables=batch_size=4064,broker_load_batch_size=16352,disable_streaming_preaggregations=false,enable_distinct_streaming_aggregation=true,parallel_fragment_exec_instance_num=3,parallel_pipeline_task_num=5,profile_level=1,enable_pipeline_engine=true,enable_parallel_scan=true,parallel_scan_max_scanners_count=48,parallel_scan_min_rows_per_scanner=16384,parallel_prepare_threshold=13,enable_fold_constant_by_be=true,enable_rewrite_element_at_to_slot=true,runtime_filter_type=12,enable_parallel_result_sink=true,sort_phase_num=0,rewrite_or_to_in_predicate_threshold=100000,enable_function_pushdown=false,enable_common_expr_pushdown=true,enable_local_exchange=false,partitioned_hash_join_rows_threshold=1048576,partitioned_hash_agg_rows_threshold=1048576,partition_pruning_expand_threshold=10,enable_share_hash_table_for_broadcast_join=false,enable_two_phase_read_opt=true,enable_common_expr_pushdown_for_inverted_index=true,enable_delete_sub_predicate_v2=false,min_revocable_mem=33554432,fetch_remote_schema_timeout_seconds=120,max_fetch_remote_schema_tablet_count=512,enable_join_spill=false,enable_sort_spill=false,enable_agg_spill=false,enable_force_spill=false,data_queue_max_blocks=1,spill_streaming_agg_mem_limit=268435456,external_agg_partition_bits=5*/ select col4 from table_7052055 limit 10;
ERROR 1105 (HY000): errCode = 2, detailMessage = [INTERNAL_ERROR][E6] nested_column's size 660, is not consistent with offsets_column's 100

	0#  doris::get_stack_trace_by_libunwind(int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
	1#  doris::get_stack_trace(int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)
	2#  doris::Exception::Exception(int, std::basic_string_view<char, std::char_traits<char> > const&)
	3#  doris::Exception::Excep
```
Issue Number: close #xxx

### CODE REVIEW

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run p0

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run p0

**Review by yiguolei — APPROVED** [review]

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by xiaokang — APPROVED** [review]
LGTM

File: be/test/vec/core/column_test.cpp [hunk 1]
```diff
@@ -118,4 +146,79 @@ TEST_F(ColumnTest, ShrinkColumnDecimal64) {
               Decimal64(4.56));
     // column decimal scale out will not empty init
 }
+
+TEST_F(ColumnTest, ShrinkColumnArray) {
```
Comment (xiaokang) [inline be/test/vec/core/column_test.cpp:150]: add testcase for expend

**Pull request merged**